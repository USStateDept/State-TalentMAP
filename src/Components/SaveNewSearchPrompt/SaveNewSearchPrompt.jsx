import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import { SAVED_SEARCH_MESSAGE, SAVED_SEARCH_OBJECT } from '../../Constants/PropTypes';
import InteractiveElement from '../InteractiveElement';
import { focusById } from '../../utilities';

const SaveNewSearchPrompt = ({ toggleInput, newSavedSearchSuccess,
  currentSavedSearch }) => {
  const currentSearchExists = currentSavedSearch.id;
  return (
    <div className="usa-grid-full">
      <div className="usa-grid-full">
        <FontAwesome name="bookmark" />
        {
          currentSearchExists ? `Saved search: ${currentSavedSearch.name}. ` : null
        }
        <InteractiveElement
          className="save-search-link"
          type="span"
          onClick={(e) => { toggleInput(e); focusById('saved-search', 1); }}
          role="link"
        >
          {currentSearchExists ? 'Edit this search.' : 'Save this search.'}
        </InteractiveElement>
        &nbsp;{currentSearchExists ? null : 'You will be able to come back to these results later.'}
      </div>
      {
        newSavedSearchSuccess &&
        <div className="usa-grid-full">
          {newSavedSearchSuccess}
        </div>
      }
    </div>
  );
};

SaveNewSearchPrompt.propTypes = {
  toggleInput: PropTypes.func.isRequired,
  newSavedSearchSuccess: SAVED_SEARCH_MESSAGE,
  currentSavedSearch: SAVED_SEARCH_OBJECT,
};

SaveNewSearchPrompt.defaultProps = {
  newSavedSearchSuccess: false,
  currentSavedSearch: {},
};

export default SaveNewSearchPrompt;
