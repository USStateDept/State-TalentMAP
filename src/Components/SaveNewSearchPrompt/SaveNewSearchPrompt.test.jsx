import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';
import toJSON from 'enzyme-to-json';
import SaveNewSearchPrompt from './SaveNewSearchPrompt';

describe('SaveNewSearchPromptComponent', () => {
  let wrapper = null;

  it('is defined', () => {
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={() => {}}
        newSavedSearchSuccess="success"
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('calls functions after clicking InteractiveElement', () => {
    const spy = sinon.spy();
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={spy}
        newSavedSearchSuccess="success"
      />,
    );
    wrapper.find('InteractiveElement').simulate('click');
    sinon.assert.calledOnce(spy);
  });

  it('can handle receiving currently selected saved searches', () => {
    const savedSearch = { id: 1, name: 'test' };
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={() => {}}
        newSavedSearchSuccess="success"
        currentSavedSearch={savedSearch}
      />,
    );
    expect(wrapper.find(savedSearch.name)).toBeDefined();
    expect(wrapper.find('Edit')).toBeDefined();
  });

  it('matches snapshot', () => {
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={() => {}}
        newSavedSearchSuccess="success"
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
