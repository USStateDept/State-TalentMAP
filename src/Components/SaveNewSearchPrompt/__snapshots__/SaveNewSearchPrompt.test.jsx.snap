// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveNewSearchPromptComponent matches snapshot 1`] = `
<div
  className="usa-grid-full"
>
  <div
    className="usa-grid-full"
  >
    <FontAwesome
      name="bookmark"
    />
    <InteractiveElement
      className="save-search-link"
      onClick={[Function]}
      role="link"
      type="span"
    >
      Save this search.
    </InteractiveElement>
    Â 
    You will be able to come back to these results later.
  </div>
  <div
    className="usa-grid-full"
  >
    success
  </div>
</div>
`;
